src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt

```
@@ -375,40 +375,39 @@ abstract class MangaDex(override val lang: String, val dexLang: String) :
         if (!helper.containsUuid(chapter.url)) {
             throw Exception("Migrate this manga from MangaDex to MangaDex to update it")
         }
-        return GET(MDConstants.apiUrl + chapter.url, headers, CacheControl.FORCE_NETWORK)
+        val chapterId = chapter.url.substringAfter("/chapter/")
+        val usingStandardHTTPS =
+            preferences.getBoolean(MDConstants.getStandardHttpsPreferenceKey(dexLang), false)
+        val atHomeRequestUrl = if (usingStandardHTTPS) {
+            "${MDConstants.apiUrl}/at-home/server/$chapterId?forcePort443=true"
+        } else {
+            "${MDConstants.apiUrl}/at-home/server/$chapterId"
+        }
+
+        return helper.mdAtHomeRequest(atHomeRequestUrl, headers, CacheControl.FORCE_NETWORK)
     }
 
     override fun pageListParse(response: Response): List<Page> {
         if (response.isSuccessful.not()) {
```

404 is considered a non successful response, so it will throw the exception instead of returning the empty list. Also, it will probably even never enter this method since there's a `asObservableSuccess()` in the `fetchPageList()` method by default, so it's probably needed to override there to handle this too.

---

Ah so its redundant and dead code now. I had a hunch but I kept it anyways. I'll just remove it completely

---

Ah wait, I think I didn't understand this fully. Were you also referring to the Error 504 that [getMdAtHomeUrl](https://github.com/tachiyomiorg/tachiyomi-extensions/blob/a478a446a325799a4ac65061680474355f7420bf/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt#L159) handles?

---

No, because in that case the response is coming directly from a client call above, so the check in this method is valid and works if needed.

---

What I was referring is that you probably need to override the `fetchPageList` to remove the `asObservableSuccess` and handle it manually to ignore 404 (and in this case returning an `Observable.just(emptyList())`)

---

Ah okay. Yeah, there wasn't a particular need to handle 404 explicitly; I was refactoring the code around it while checking the api docs.
After looking at it again, I'll just let `fetchPageList` handle it like it normally does. Thanks for catching the dead code!